### PR TITLE
LIVE-1625 Add hide NFT collection feature

### PR DESCRIPTION
**Pull request by grsoares21**
This PR adds the possibility of hiding NFT collections from both the account page and the NFT Gallery page.
The hidden collections are visible in the account settings page and can be unhidden there.

https://user-images.githubusercontent.com/6013294/159961708-04afeea7-3f4a-45e4-8543-f53a42153125.mp4

### Type
Feature

### Context
[LIVE-1625]

### Parts of the app affected / Test plan
Account page, NFT Gallery and Setting page.
On an account page, long-press an NFT collection to show the hide menu.
On the NFT Gallery, click the 3 dots beside the collection title to show the hide menu.
Go to settings -> accounts -> hidden NFT collections to unhide the collections.
Hidden collections should not be visible in the account page nor the NFT gallery.

[LIVE-1625]: https://ledgerhq.atlassian.net/browse/LIVE-1625?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Nft/NftCollectionOptionsMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          <Button type="default" style={styles.confirmDrawerButtons} onPress={() => setIsConfirmDrawerOpen(false)}>
+            {t("common.cancel")}
+          </Button>
+        </View>
+      </BottomDrawer>
+    </>
+  );
+};
+
+const styles = StyleSheet.create({
```
Comment (LFBarreto) [inline src/components/Nft/NftCollectionOptionsMenu.tsx:88]: No need to use StyleSheet if you use native-ui components they implement all of these styles as styled-component props 
```js
<View style={styles.menuItem}>
->
<Flex flexDirection="row" alignItems="center" >
```

File: src/screens/Settings/Accounts/HiddenNftCollections.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        data={hiddenCollections}
+        renderItem={renderItem}
+        keyExtractor={keyExtractor}
+        contentContainerStyle={styles.containerStyle}
+      />
+    </Box>
+  );
+};
+
+const styles = StyleSheet.create({
```
Comment (LFBarreto) [inline src/screens/Settings/Accounts/HiddenNftCollections.tsx:90]: same here you dont need StyleSheet anymore if you need to use styles you cant target via props use styled and extend on it 
```js
import styled from "styled-components/native";

const StyledNFTImage = styled(NFTImage)`
    borderRadius: 4,
    width: 36,
    aspectRatio: 1,
    overflow: "hidden",
`
```

**Review by LFBarreto — APPROVED** [review]

**Review by LFBarreto — CHANGES_REQUESTED** [review]
some changes to be made if you use new styling tools :+1:

File: src/screens/Settings/Accounts/HiddenNftCollections.tsx (hunk 2 above)
Comment (grsoares21) [reply src/screens/Settings/Accounts/HiddenNftCollections.tsx]: Done

File: src/components/Nft/NftCollectionOptionsMenu.tsx (hunk 1 above)
Comment (grsoares21) [reply src/components/Nft/NftCollectionOptionsMenu.tsx]: Done

**Comment by lambertkevin** [comment]
Closed for the rebased version here: https://github.com/LedgerHQ/ledger-live-mobile/pull/2452

**Pull request closed without merging**